From 638681cd91533bb2a6837f31507ae7ccf0950332 Mon Sep 17 00:00:00 2001
From: Steve Kipisz <s-kipisz2@ti.com>
Date: Tue, 14 Jun 2011 12:12:33 -0500
Subject: [PATCH 2/2] Increase MPU clock to 1GHz

Signed-off-by: Steve Kipisz <s-kipisz2@ti.com>
---
 board/omap3evm/platform.S |    8 ++++----
 1 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/board/omap3evm/platform.S b/board/omap3evm/platform.S
index 1453307..dba9317 100644
--- a/board/omap3evm/platform.S
+++ b/board/omap3evm/platform.S
@@ -437,10 +437,10 @@ get_per_dpll_param:
  * Tables for 36x/37x devices
  */
 mpu_36x_dpll_param:
-.word 300,  5, 0, 1	// 12MHz
-.word 600, 12, 0, 1	// 13MHz
-.word 125,  3, 0, 1	// 19.2MHz
-.word 300, 12, 0, 1	// 26MHz
+.word  250,  2, 0, 1	// 12MHz
+.word 1000, 12, 0, 1	// 13MHz
+.word 1875, 35, 0, 1	// 19.2MHz
+.word  500, 12, 0, 1	// 26MHz
 
 iva_36x_dpll_param:
 .word 130,  2, 0, 1
-- 
1.7.2.3

