From ce01d6622d05069e24c453c4b6461d94edd81bcd Mon Sep 17 00:00:00 2001
From: don.darling <don.darling@24075187-2e39-4e88-bbb8-bc8aa768f540>
Date: Fri, 30 Jul 2010 15:46:12 +0000
Subject: [PATCH 1/8] Add build support for DM365 on DVSDK 4.00.

---
 gstreamer_ti/Makefile.common |   13 +++++++------
 1 files changed, 7 insertions(+), 6 deletions(-)

diff --git a/gstreamer_ti/Makefile.common b/gstreamer_ti/Makefile.common
index 8ea469b..d9572b8 100644
--- a/gstreamer_ti/Makefile.common
+++ b/gstreamer_ti/Makefile.common
@@ -174,12 +174,13 @@ ifeq ($(GST_TI_PLATFORM), dm6446)
 endif
 
 ifeq ($(GST_TI_PLATFORM), dm365)
-    export XDC_TARGET    = gnu.targets.arm.GCArmv5T
-    export XDC_PLATFORM  = ti.platforms.evmDM365
-    export MVTOOL_PREFIX = $(CSTOOL_PREFIX)
-    export MVTOOL_DIR    = $(CSTOOL_DIR)
-    export PLATFORM_XDC  = ${XDC_PLATFORM}
-    export CROSS_COMPILE = ${CSTOOL_PREFIX}
+    export XDC_TARGET           = gnu.targets.arm.GCArmv5T
+    export XDC_PLATFORM         = ti.platforms.evmDM365
+    export MVTOOL_PREFIX        = $(CSTOOL_PREFIX)
+    export MVTOOL_DIR           = $(CSTOOL_DIR)
+    export PLATFORM_XDC         = ${XDC_PLATFORM}
+    export CROSS_COMPILE        = ${CSTOOL_PREFIX}
+    export LIBTOOL_SYSROOT_PATH = $(LINUXLIBS_INSTALL_DIR)/..
 endif
 
 ifeq ($(GST_TI_PLATFORM), dm6467)
-- 
1.7.0.4

