Index: matrix-gui-e/ti816x/etc/init
===================================================================
--- matrix-gui-e/ti816x/etc/init	(revision 232)
+++ matrix-gui-e/ti816x/etc/init	(working copy)
@@ -2,6 +2,7 @@
 
 matrixgui="/usr/bin/matrix_guiE"
 GUI_OPTS="-qws /usr/share/matrix/html/menu_main.html"
+MULTI_OPTS="Multi: LinuxFb:/dev/fb0 LinuxFb:/dev/fb1"
 PIDFILE="/var/run/matrix-gui-e.pid"
 
 test -x "$matrixgui" || exit 0
@@ -29,7 +30,7 @@
     fi
 
     echo -n "Starting Matrix GUI application"
-    start-stop-daemon --start --quiet --background --exec $matrixgui -- $GUI_OPTS
+    start-stop-daemon --start --quiet --background --exec $matrixgui -- -display "$MULTI_OPTS" $GUI_OPTS
 	pidof matrix_guiE > $PIDFILE
     echo "."
     ;;
