From e2078a51df00736cf09b3cdb57c00daa55aabfde Mon Sep 17 00:00:00 2001
From: Amarinder Bindra <a-bindra@ti.com>
Date: Tue, 31 Jan 2012 14:51:42 +0530
Subject: [PATCH 5/5] ti814x ti816x: changes in Makefile to compile the files for logo


Signed-off-by: Amarinder Bindra <a-bindra@ti.com>
---
 drivers/video/Makefile |    6 ++++++
 1 files changed, 6 insertions(+), 0 deletions(-)

diff --git a/drivers/video/Makefile b/drivers/video/Makefile
index 7d84fc7..55a467c 100644
--- a/drivers/video/Makefile
+++ b/drivers/video/Makefile
@@ -38,6 +38,12 @@ COBJS-$(CONFIG_SED156X) += sed156x.o
 COBJS-$(CONFIG_VIDEO_SM501) += sm501.o
 COBJS-$(CONFIG_VIDEO_SMI_LYNXEM) += smiLynxEM.o videomodes.o
 COBJS-$(CONFIG_VIDEO_VCXK) += bus_vcxk.o
+ifdef CONFIG_TI816X
+COBJS-$(CONFIG_VIDEO) += ti816x_logo.o firmware.o
+endif
+ifdef CONFIG_TI814X
+COBJS-$(CONFIG_VIDEO) += ti814x_logo.o firmware.o
+endif
 
 COBJS	:= $(COBJS-y)
 SRCS	:= $(COBJS:.o=.c)
-- 
1.7.1

