From 8167799941fa48b8faf5d8ee11ac028ad7eea806 Mon Sep 17 00:00:00 2001
From: Amarinder Bindra <a-bindra@ti.com>
Date: Thu, 2 Feb 2012 16:29:23 +0530
Subject: [PATCH] ti816x: adding config CONFIG_TI81XX_VIDEO for logo


Signed-off-by: Amarinder Bindra <a-bindra@ti.com>
---
 Makefile                     |    1 +
 include/configs/ti8168_evm.h |   16 +++++++++-------
 2 files changed, 10 insertions(+), 7 deletions(-)

diff --git a/Makefile b/Makefile
index b56b11e..5bc195f 100644
--- a/Makefile
+++ b/Makefile
@@ -3320,6 +3320,7 @@ ti8168_evm_min_sd:	unconfig
 	else	\
 		echo "#define CONFIG_SYS_NO_FLASH"    >>$(obj)include/config.h ; \
 		echo "#define CONFIG_NAND_ENV"    >>$(obj)include/config.h ; \
+		echo "#define CONFIG_TI81XX_VIDEO"    >>$(obj)include/config.h ; \
 		echo "Setting up TI8168 default build with NAND..." ; \
 	fi;
 	@$(MKCONFIG) -a ti8168_evm arm arm_cortexa8 ti8168 ti ti81xx
diff --git a/include/configs/ti8168_evm.h b/include/configs/ti8168_evm.h
index 0d27240..f451806 100644
--- a/include/configs/ti8168_evm.h
+++ b/include/configs/ti8168_evm.h
@@ -294,13 +294,15 @@ extern unsigned int boot_flash_type;
 #endif	/* NOR support */
 
 /* Splashscreen support */
-#define CONFIG_VIDEO
-#define CONFIG_CFB_CONSOLE
-#define CONFIG_VIDEO_LOGO
-#define CONFIG_VGA_AS_SINGLE_DEVICE
-#define CONFIG_SYS_CONSOLE_IS_IN_ENV
-#define CONFIG_CMD_BMP
-#define VIDEO_FB_LITTLE_ENDIAN
+#ifdef CONFIG_TI81XX_VIDEO
+# define CONFIG_VIDEO
+# define CONFIG_CFB_CONSOLE
+# define CONFIG_VIDEO_LOGO
+# define CONFIG_VGA_AS_SINGLE_DEVICE
+# define CONFIG_SYS_CONSOLE_IS_IN_ENV
+# define CONFIG_CMD_BMP
+# define VIDEO_FB_LITTLE_ENDIAN
+#endif
 
 /* I2C support */
 #ifdef CONFIG_I2C
-- 
1.7.1

