diff -uNr demos_orig/dm365/loadmodules.sh demos/dm365/loadmodules.sh
--- demos_orig/dm365/loadmodules.sh	2010-09-03 17:39:10.000000000 -0500
+++ demos/dm365/loadmodules.sh	2010-09-12 11:12:14.241160129 -0500
@@ -1,4 +1,4 @@
-#
+#!/bin/sh
 # loadmodules configuring CMEM for DVSDK demos on DM365 platform
 #
 # Copyright (C) 2010 Texas Instruments Incorporated - http://www.ti.com/ 
@@ -32,8 +32,6 @@
 #  (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE 
 #  OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 #
-#!/bin/sh
-
 # Default Memory Map
 #
 # Start Addr    Size    Description
@@ -41,18 +39,8 @@
 # 0x80000000    60 MB   Linux
 # 0x83C00000    68 MB   CMEM
 
-# Sample bootargs is given below
-# mem=60M console=ttyS0,115200n8 root=/dev/nfs rw nfsroot=<nfsroot> ip=dhcp video=davincifb:vid0=OFF:vid1=OFF:osd0=720x576x16,2025K dm365_imp.oper_mode=0
-
+# remove previously loaded cmem to ensure that its using our cmem pool
 rmmod cmemk 2>/dev/null
-rmmod irqk 2>/dev/null
-rmmod edmak 2>/dev/null
-rmmod dm365mmap 2>/dev/null
-
-depmod -a cmemk
-depmod -a irqk
-depmod -a dm365mmap
-depmod -a edmak
 
 # Pools configuration
 modprobe cmemk phys_start=0x83C00000 phys_end=0x88000000 pools=2x384,1x16384,1x2688,1x20480,3x74,1x60288,2x28,1x147048,1x10240,1x896,1x65536,1x663552,1x9175376,4x24,1x282624,2x5984,1x58144,1x1024,1x48952,1x464,2x16768,1x1251264,1x30720,1x65792,8x774144,1x173712,1x146,1x98,2x296,50x56,2x86,1x624,1x3328,1x518912,4x62,1x242,4x663552,1x1637824,14x1645056,1x16538976,4x1382400
@@ -63,3 +51,4 @@
 
 rm -f /dev/dm365mmap
 mknod /dev/dm365mmap c `awk "\\$2==\"dm365mmap\" {print \\$1}" /proc/devices` 0
+
