diff -uNr demos_orig/omap3530/loadmodules.sh demos/omap3530/loadmodules.sh
--- demos_orig/omap3530/loadmodules.sh	2010-09-01 14:46:47.000000000 -0500
+++ demos/omap3530/loadmodules.sh	2010-09-12 11:06:37.149160036 -0500
@@ -1,6 +1,4 @@
 #!/bin/sh
-depmod -a
-
 #
 # Default Memory Map
 #
@@ -10,6 +8,8 @@
 # 0x86300000     15 MB  CMEM
 # 0x87200000     13 MB  CODEC SERVER
 
+# remove previously loaded cmem to ensure that its using our cmem pool.
+rmmod cmemk  2>/dev/null
 
 # Allocate 15MB for CMEM
 modprobe cmemk phys_start=0x86300000 phys_end=0x87200000 pools=1x3000000,1x1429440,6x1048576,4x829440,1x327680,1x256000,7x131072,20x4096 allowOverlap=1
