From 4c12b10bbddb3656ee16483731e46a0b05df7b8c Mon Sep 17 00:00:00 2001
From: Amarinder Bindra <a-bindra@ti.com>
Date: Thu, 9 Feb 2012 10:02:14 +0530
Subject: [PATCH 4/4] ti816x: added CONFIG_TI81XX_VIDEO for NAND


Signed-off-by: Amarinder Bindra <a-bindra@ti.com>
---
 Makefile |    1 +
 1 files changed, 1 insertions(+), 0 deletions(-)

diff --git a/Makefile b/Makefile
index 02826c3..5aeeac3 100644
--- a/Makefile
+++ b/Makefile
@@ -3299,6 +3299,7 @@ ti8168_evm_min_sd:	unconfig
 	@if [ "$(findstring _nand,$@)" ] ; then \
 		echo "#define CONFIG_SYS_NO_FLASH"    >>$(obj)include/config.h ; \
 		echo "#define CONFIG_NAND_ENV"    >>$(obj)include/config.h ; \
+		echo "#define CONFIG_TI81XX_VIDEO"    >>$(obj)include/config.h ; \
 		echo "Setting up TI8168 NAND build with ENV in NAND..." ; \
 	elif [ "$(findstring _nor,$@)" ] ; then \
 		echo "#define CONFIG_NOR"    >>$(obj)include/config.h ; \
-- 
1.7.1

