diff -Naur a/arch/arm/mach-omap2/board-ti8168evm.c b/arch/arm/mach-omap2/board-ti8168evm.c
--- a/linux-04.00.00.10/arch/arm/mach-omap2/board-ti8168evm.c	2011-02-15 03:07:23.000000000 +0530
+++ b/arch/arm/mach-omap2/board-ti8168evm.c	2011-04-12 23:46:06.478179000 +0530
@@ -491,7 +491,7 @@
 
 static void __init ti816x_vpss_init(void)
 {
-	i2c_add_driver(&pcf8575_driver);
+	//i2c_add_driver(&pcf8575_driver);
 
 	if (platform_device_register(&vpss_device))
 		printk(KERN_ERR "failed to register ti816x_vpss device\n");
diff -Naur a/linux-04.00.00.10/drivers/video/ti81xx/vpss/core.c b/drivers/video/ti81xx/vpss/core.c
--- a/linux-04.00.00.10/drivers/video/ti81xx/vpss/core.c	2011-02-15 03:07:23.000000000 +0530
+++ b/drivers/video/ti81xx/vpss/core.c	2011-04-12 23:46:27.954024000 +0530
@@ -87,11 +87,13 @@
 		goto exit3;
 
 	}
+#if 0
 	r = vps_video_init(pdev);
 	if (r) {
 		VPSSERR("failed to int video.\n");
 		goto exit4;
 	}
+#endif
 	return 0;
 
 exit4:
@@ -111,9 +113,9 @@
 static int vps_remove(struct platform_device *pdev)
 {
 	int r;
-
+#if 0
 	vps_video_deinit(pdev);
-
+#endif
 	vps_grpx_deinit(pdev);
 	r = vps_dc_deinit(pdev);
 	if (r) {
diff -Naur a/linux-04.00.00.10/drivers/video/ti81xx/vpss/dctrl.c b/drivers/video/ti81xx/vpss/dctrl.c
--- a/linux-04.00.00.10/drivers/video/ti81xx/vpss/dctrl.c	2011-02-15 03:07:23.000000000 +0530
+++ b/drivers/video/ti81xx/vpss/dctrl.c	2011-04-12 23:46:46.669063000 +0530
@@ -1303,6 +1303,7 @@
 #ifdef CONFIG_ARCH_TI816X
 	if (cpu_is_ti816x()) {
 		if ((binfo->idx == HDCOMP) && (binfo->isdeviceon == true)) {
+                        #if 0
 			if ((mid == FVID2_STD_1080P_60) ||
 			    (mid == FVID2_STD_1080P_50))
 				r = pcf8575_ths7360_hd_enable(
@@ -1314,6 +1315,7 @@
 				VPSSERR("failed to set THS filter\n");
 				goto exit;
 			}
+                        #endif
 
 		}
 	}
@@ -2541,6 +2543,7 @@
 	if setup is failed*/
 #ifdef CONFIG_ARCH_TI816X
 	if (cpu_is_ti816x()) {
+                #if 0
 		r = pcf8575_ths7375_enable(TI816X_THSFILTER_ENABLE_MODULE);
 		if ((venc_info.modeinfo[HDCOMP].minfo.standard ==
 		    FVID2_STD_1080P_60)  ||
@@ -2562,6 +2565,7 @@
 			disp_ctrl->blenders[SDVENC].isdeviceon = false;
 
 		}
+                #endif
 	}
 #endif
 	return 0;
